### PR TITLE
Replace easing() function with tokens

### LINKED ISSUE #4595: Move easing() and duration() function values to tokens

### LINKED ISSUE #4597: Replace easing() function with tokens

### PR DESCRIPTION
**Pull request by lgriffee**
### WHY are these changes introduced?

Fixes #4597 

To deprecate the `easing()` function from our public sass API and replace it with tokens. We no longer want to support this functionality in the system.

### WHAT is this pull request doing?

- Removing all instances of `easing()`
- Updating the `--p-ease` token value
- Updating the `--p-ease-in` token value
- Adding a new `--p-ease-out` token
- Adding a new `--p-ease-excite` token

Note:
`easing(overshoot)` and `easing(anticipate)` were not moved to token values since they are not used in `polaris-react` or `web`—they have been completely removed.

### CODE REVIEW

**Comment by lgriffee** [comment]
I'm noticing that Chromatic is detecting UI changes but I'm not sure what is causing those yet...The same UI tests also seem to be affected in #4699

File: UNRELEASED.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
 
 ### Breaking changes
 
+- Removed the easing() function and replaced any instances with tokens ([#4698](https://github.com/Shopify/polaris-react/pull/4698))
+
```
Comment (alex-page) [inline UNRELEASED.md]: This should go into `UNRELEASED-v8` file.

**Review by alex-page — APPROVED** [review]

File: src/components/Sheet/Sheet.scss [hunk 2]
```diff
@@ -40,7 +40,7 @@ $sheet-desktop-width: rem(380px);
 
 .Bottom {
   will-change: transform;
-  transition: transform duration('slow') easing('base');
+  transition: transform duration('slow') --p-ease;
```
Comment (aaronccasanova) [inline src/components/Sheet/Sheet.scss]:
```suggestion
  transition: transform duration('slow') var(--p-ease);
```

File: src/components/Sheet/Sheet.scss [hunk 3]
```diff
@@ -58,7 +58,7 @@ $sheet-desktop-width: rem(380px);
 
 .Right {
   will-change: transform;
-  transition: transform duration('slow') easing('base');
+  transition: transform duration('slow') --p-ease;
```
Comment (aaronccasanova) [inline src/components/Sheet/Sheet.scss]:
```suggestion
  transition: transform duration('slow') var(--p-ease);
```

**Review by aveline — APPROVED** [review]

**Review by aaronccasanova — APPROVED** [review]

**Pull request merged**